src/main/java/com/googlecode/download/maven/plugin/internal/WGet.java

```diff
@@ -230,6 +237,12 @@
     @Parameter(property = "checkSignature", defaultValue = "false")
     private boolean checkSignature;
 
+    /**
+     * A list of additional HTTP headers to send with the request
+     */
+    @Parameter(property = "download.plugin.headers")
+    private List<String> headers = new ArrayList<>();
```

Comment:
@albx79 would it make sense to have a Map here?

Reply:
I don't know, compared to a simple "header-name: value", I think the map is just adding syntactic noise. Apache Http Client provides a class to parse the colon-separated string, so I don't thing a map would buy us much.

Reply:
@albx79 the main benefit of a Map is that we don't need to parse anything at all. The only thing we would need it construct `Header` instances from K/V pairs.
Config example
```xml
<headers>
	<X-Custom-1>first custom header</X-Custom-1>
	<X-Custom-2>second custom header</X-Custom-2>
</headers>
```
And `getAdditionalHeaders()` becomes as simple as this
```java
        return this.headers.entrySet().stream().map(
            pair -> new BasicHeader(pair.getKey(), pair.getValue())
        ).collect(Collectors.toList());
```
Make sense?